js_modules/dagster-ui/packages/ui-core/src/performance/TraceContext.tsx

```diff
@@ -0,0 +1,107 @@
+import {QueryResult} from '@apollo/client';
+import {ReactNode, createContext, useContext, useLayoutEffect, useMemo} from 'react';
+
+type TraceContextType = {
+  createDependency: (_name: string) => Dependency | null;
+  addDependency: (_dep: Dependency | null) => void;
+  cancelDependency: (_dep: Dependency | null) => void;
+  completeDependency: (_dep: Dependency | null) => void;
+  completeDependencyWithError: (_dep: Dependency | null) => void;
+};
+
+export const TraceContext = createContext<TraceContextType>({
+  createDependency: (_name: string) => null,
+  addDependency: (_dep) => {},
+  cancelDependency: (_dep) => {},
+  completeDependency: (_dep) => {},
+  completeDependencyWithError: () => {},
```

Comment:
this api reads a little weird, what led you to this versus one `completeDependency` with argument(s) ? maybe tweaking the names to make the two complete methods feel like equal alternative options

Reply:
Ah yeah I didn't really consider that. I don't feel strongly either way tbh.

Reply:
i think avoiding the possibility of calling both different on completes is my best argument for one